WordPress/gutenberg
# Site Editor - add basic plugin support

## Description


Adds basic plugin support to the Site Editor with edit-site specific version of:
* PluginSidebar
* PluginSidebarMoreMenuItem
* PluginMoreMenuItem
* openGeneralSidebar / closeGeneralSidebar store actions

## How has this been tested?




Tested by registering a plugin with these new components and using them in the Site Editor.

### PluginSidebar

Register the plugin and add the sidebar:

```
import { registerPlugin } from '@wordpress/plugins';
import { PluginSidebar } from '@wordpress/edit-site';
import { comment } from '@wordpress/icons';

const PluginSidebarTest = () => (
	<PluginSidebar
		name="plugin-sidebar-test"
		title="My Plugin"
		icon={ comment }
	>
		<p>Plugin Sidebar</p>
	</PluginSidebar>
);

registerPlugin( 'plugin-sidebar-test', { render: PluginSidebarTest } );
```

* Load the site editor and verify the plugin icon shows up in the header and its corresponding menu item in the more menu.

![Screen Shot 2021-09-01 at 2 16 11 PM](https://user-images.githubusercontent.com/28742426/131723306-ca6dcb59-91b0-4c8e-baa7-f7b4d13edae6.png)


* Verify these buttons toggle the plugin sidebar.
* Add the `isPinnable={ false }` prop to the `PluginSidebar` component in the example above.  Reload the editor and verify neither of these buttons are present.



### PluginSidebarMoreMenuItem & PluginMoreMenuItem

Register the plugin, pass the `isPinnalbe={ false }` prop to the `PluginSidebar` component, and register the two more menu items.

```
import { registerPlugin } from '@wordpress/plugins';
import {
	PluginSidebar,
	PluginSidebarMoreMenuItem,
	PluginMoreMenuItem,
} from '@wordpress/edit-site';
import { comment } from '@wordpress/icons';

const PluginSidebarTest = () => (
	<>
		<PluginSidebar
			name="plugin-sidebar-test"
			title="My Plugin"
			icon={ comment }
			isPinnable={ false }
		>
			<p>Plugin Sidebar</p>
		</PluginSidebar>
		<PluginSidebarMoreMenuItem
			target="plugin-sidebar-test"
			icon={ comment }
		>
			Sidebar Menu Item
		</PluginSidebarMoreMenuItem>
		<PluginMoreMenuItem icon={ comment }>
			Plugin menu item
		</PluginMoreMenuItem>
	</>
);

registerPlugin( 'plugin-sidebar-test', { render: PluginSidebarTest } );
```

* Load the site editor and verify that there is no button for the plugin in the header.  
* Verify that there is the `PluginSidebarMoreMenuItem` exists in the more menu (labeled "Sidebar Menu Item" in our example code above) and that clicking on this opens the plugin sidebar. 
* Also verify that the `PluginMoreMenuItem` exists in the more menu (labeled "Plugin menu item" in the example code above), although clicking should have no effect as we have not set any link or button action for the plugin.

![Screen Shot 2021-09-01 at 2 22 11 PM](https://user-images.githubusercontent.com/28742426/131723683-e49fb006-04a2-4912-b441-47a3db63a1a7.png)

### openGeneralSidebar / closeGeneralSidebar actions

With the plugin registered and site editor loaded, open the console window and verify that the following actions open and close the plugin's sidebar as expceted:
* `wp.data.dispatch( 'core/edit-site' ).openGeneralSidebar( 'plugin-sidebar-test/plugin-sidebar-test' )`
* `wp.data.dispatch( 'core/edit-site' ).closeGeneralSidebar( 'plugin-sidebar-test/plugin-sidebar-test' )`

## Screenshots 

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
